paddle/fluid/imperative/gradient_accumulator.cc

```diff
@@ -133,8 +185,34 @@ void TensorAdd(const framework::Variable& src, framework::Variable* dst) {
 
 #undef PADDLE_TENSOR_ADD
 
-  PADDLE_THROW("Not supported data type %s for AddTo",
-               framework::DataTypeToString(data_type));
+  if (data_type == framework::proto::VarType::FP16) {
+    if (platform::is_gpu_place(place)) {
+#ifdef PADDLE_WITH_CUDA
+      TensorAddImpl<platform::CUDADeviceContext, platform::float16>(
+          src_tensor, dst_tensor, place);
+#else
+      PADDLE_THROW(platform::errors::PermissionDenied(
```

Comment:
may be `Unimplemented` is better than `PermissionDenied` here?

Reply:
I will change that.